Make `snd_sof_ipc_set_get_comp_data()` IPC version agnostic

ujfalusi opened the pull request:
Hi,

currently the `snd_sof_ipc_set_get_comp_data()` have parameters directly referring to IPC types, definitions.
Drop them with the exception of `enum sof_ipc_ctrl_type ctrl_type` which is kind of a mystery parameter itself as for example the `SOF_CTRL_TYPE_VALUE_COMP_GET` / `SOF_CTRL_TYPE_VALUE_COMP_SET` is not used on kernel side and it is not at all defined what it would require to support it.

I was tempted to drop them as such and use only `SOF_CTRL_TYPE_VALUE_CHAN_*` / `SOF_CTRL_TYPE_DATA_*` (as the code only uses these, but it is so confusing that I was not comfortable doing so.

Edit: (as with v2): the `SOF_CTRL_TYPE_VALUE_COMP_GET` / `SOF_CTRL_TYPE_VALUE_COMP_SET` is not longer handled by the kernel, it is not used on firmware side either.

plbossart commented on sound/soc/sof/control.c:
```diff
@@ -137,9 +136,7 @@ int snd_sof_volume_put(struct snd_kcontrol *kcontrol,
-					      SOF_CTRL_TYPE_VALUE_CHAN_SET,
-					      SOF_CTRL_CMD_VOLUME,
-					      true);
+					      SOF_CTRL_TYPE_VALUE_CHAN_SET, true);
```
typo in commit message:  s/scrontrol/scontrol

ujfalusi replied:
OK

kv2019i replied:
Maybe also correct "duplication information" to "duplicated information"

ujfalusi replied:
The commit messages went through several iterations and you are right, I'll fix this up as well.
Thanks!

plbossart commented on sound/soc/sof/control.c:
```diff
@@ -69,7 +69,7 @@ static void snd_sof_refresh_control(struct snd_sof_control *scontrol)
 	struct sof_ipc_ctrl_data *cdata = scontrol->control_data;
 	struct snd_soc_component *scomp = scontrol->scomp;
-	enum sof_ipc_ctrl_type ctrl_type;
+	u32 ipc_cmd;
```
missing space in commit title: ASoC: SOF: control: Fix variable type in snd_sof_refresh_control()

ujfalusi replied:
OK.

plbossart reviewed (changes requested):
2 nit-picks.

ujfalusi commented:
Changes since v1:
- Fixed the two typos pointed out by @plbossart 
- Do not handle SOF_CTRL_TYPE_VALUE_COMP_* type for controls as it is not used (in firmware and in kernel) and it is not clear how it supposed to be handled.
- With this change we can drop the ctrl_type parameter also from `snd_sof_ipc_set_get_comp_data()` to make it completely IPC agnostic

kv2019i reviewed (approved):
Looks good. One additional typo but it gets the message through so no need to update for this.

@juimonen  Could you look at this as well?

ujfalusi commented:
Changes since v2:
- another typo fixed, thanks to @kv2019i

ranj063 commented on sound/soc/sof/control.c:
```diff
@@ -577,6 +577,13 @@ void snd_sof_control_notify(struct snd_sof_dev *sdev,
+		dev_err(sdev->dev,
+			"Component data is not supported in control notification\n");
+		return;
+	}
```
is this correct @ujfalusi ? if it is then we don't need the cases in the switch below on line 653?

ujfalusi replied:
We also have cases in other place in ipc.c and another in topology.c.

I thought to keep them, just in case if we have something emerging which would need it (I still can not figure out how it is supposed to wotk).

I can drop all of them in next version.

ujfalusi commented:
Changes since v3:
- Drop handling of SOF_CTRL_TYPE_VALUE_COMP_* entirely as suggested by @ranj063

ranj063 reviewed (approved):
lgtm. thanks @ujfalusi

plbossart commented:
@kv2019i you good with the updated version?
